sql/core/src/main/scala/org/apache/spark/sql/SparkSession.scala

```diff
@@ -897,6 +898,18 @@ object SparkSession extends Logging {
       this
     }
 
+    // These configurations related to driver when deploy like `spark.master`,
+    // `spark.driver.memory`, this kind of properties may not be affected when
+    // setting programmatically through SparkConf in runtime, or the behavior is
+    // depending on which cluster manager and deploy mode you choose, so it would
+    // be suggested to set through configuration file or spark-submit command line options.
+    private val DRIVER_RELATED_LAUNCHER_CONFIG = Seq(DRIVER_MEMORY, DRIVER_CORES.key,
+      DRIVER_MEMORY_OVERHEAD.key, DRIVER_EXTRA_CLASSPATH,
+      DRIVER_DEFAULT_JAVA_OPTIONS, DRIVER_EXTRA_JAVA_OPTIONS, DRIVER_EXTRA_LIBRARY_PATH,
+      "spark.driver.resource", PYSPARK_DRIVER_PYTHON, PYSPARK_PYTHON, SPARKR_R_SHELL,
+      CHILD_PROCESS_LOGGER_NAME, CHILD_CONNECTION_TIMEOUT, DRIVER_USER_CLASS_PATH_FIRST.key,
+      "spark.yarn.*")
```

Comment:
@AngersZhuuuu, are they all configurations to include? is it just a subset of them?

Reply:
> @AngersZhuuuu, are they all configurations to include? is it just a subset of them?

That's what I found through configuration page. I am not sure if there is configurations in code that I missed.

Reply:
It seems like something we should build into our config library. It's hard to maintain this list here.

Reply:
> It seems like something we should build into our config library. It's hard to maintain this list here.

You mean like what  I have said in https://github.com/apache/spark/pull/31598#discussion_r581689593.
We should add a new tag in ConfigEntry? like `.version()` we add `.type()` or `.scope()` ?

Reply:
yea, something like `.scope`.

Reply:
> yea, something like `.scope`.

yea. should we do it in this pr or a new one? since add tag and check is different things

Reply:
We can do that in a new PR, and merge that PR first.

Reply:
> We can do that in a new PR, and merge that PR first.

Ok, I think it's a very usage improvement, since many user don't understand spark's configuration very well.